Add nfs-server-provisioner helm chart

kvaps opened the pull request:
Moving helm chart from [stable/nfs-server-provisioner](https://github.com/helm/charts/tree/master/stable/nfs-server-provisioner) and [kvaps/nfs-server-provisioner-chart](https://github.com/kvaps/nfs-server-provisioner-chart) repositories.

All the history of commits is preserved

kiall committed:
[stable/nfs-server-provisioner]: Add nfs-server-provisioner chart (#3870)

* [stable/nfs-provisioner]: Add nfs-provisioner chart

Adds a new chart to deploy the kubernetes-incubator nfs-provisioner
using a StatefulSet. Additionally, it requested, will define the
necessary StorageClass resource needed to make use of the provisioner.

* [stable/nfs-provisioner]: Update image tag to v1.0.9

* [stable/nfs-server-provisioner]: Move stable/nfs-provisioner to stable/nfs-server-provisioner

* [stable/nfs-server-provisioner]: Update Chart.yaml and README.md for name change

vtuson committed:
nfs service name did not match to env variable in statefulset (#5272)

* nfs service name did not match to env variable in statefulset

* version bump

* updating home and sources

* home in one line

* revert source link change

AdamDang committed:
[stable/nfs-server-provisioner]typo fix: resoure->resource (#5289)

* Add version for nfs-server-provisioner

version: 0.1.2

* [stable/nfs-server-provisioner]typo fix: resoure->resource

[stable/nfs-server-provisioner]typo fix: resoure->resource

Commit:
[stable/nfs-server-provisioner] Fix error related to `printf` in helpers (#5376)

* Fix error related to `printf` in helpers

Error: render error in "../charts/nfs-server-provisioner/templates/storageclass.yaml": template: ../charts/nfs-server-provisioner/templates/_helpers.tpl:39:4: executing "nfs-provisioner.provisionerName" at <printf>: wrong number of args for printf: want at least 1 got 0

* Bump version to 0.1.3

AdamDang committed:
[stable/nfs-server-provisioner]Typo fix: tables lists/configration (#5716)

* Add version: 0.1.4

Add version: 0.1.4

* [stable/nfs-server-provisioner]Typo fix: tables lists/configration

[stable/nfs-server-provisioner]Typo fix: tables lists/configration

mirake committed:
Typo fix: configration -> configuration (#6210)

alebabai committed:
[stable/nfs-server-provisioner]: add reclaim policy support (#7604)

* feat(stable.nfs-server-provisioner): add recliamPolicy

Signed-off-by: Alexander Babai <aliaksandr.babai@gmail.com>

* chore(stable.nfs-server-provisioner): bump chart version

Signed-off-by: Alexander Babai <aliaksandr.babai@gmail.com>

joaocc committed:
[stable/nfs-server-provisioner] Moved mountOptions to top level (#8972)

* mountOptions should be on top level

see https://github.com/kubernetes-incubator/external-storage/blob/master/nfs/docs/usage.md

Signed-off-by: Joao C Costa <joaocc-dev@live.com>

* moved mountOptions to top level, as per https://github.com/kubernetes-incubator/external-storage/blob/master/nfs/docs/usage.md

Signed-off-by: Joao C Costa <joaocc-dev@live.com>

* Fix the fix (mountOptions was in the wrong place)

Signed-off-by: Joao C Costa <joaocc-dev@live.com>

* added noatime as default mount option

Signed-off-by: Joao C Costa <joaocc-dev@live.com>

joaocc committed:
Add support for dynamically resizeable PVC (#11561)

Signed-off-by: Joao C Costa <joaocc-dev@live.com>

kim0 committed:
Upgrade nfs-provisioner to v2.2.1-k8s1.12 (#13140)

Signed-off-by: Ahmed Kamal <email.ahmedkamal@googlemail.com>

javsalgar committed:
[stable/nfs-server-provisioner] Add compatibility with k8s 1.16 (#17547)

Signed-off-by: Javier J. Salmeron Garcia <jsalmeron@vmware.com>

Commit:
Update image (#19842)

Signed-off-by: Naseem <naseemkullah@gmail.com>

Commit:
Add extraArgs and myself as owner (#20371)

Signed-off-by: Naseem <naseem@transit.app>

kvaps committed:
[stable/nfs-server-provisioner] New nfs-provisioner with NFSv3 support (#19260)

* Enable NFSv3 support, set NFSv3 by default

Signed-off-by: kvaps <kvapss@gmail.com>

* remove relatime

Signed-off-by: kvaps <kvapss@gmail.com>

* Use nindents

Signed-off-by: kvaps <kvapss@gmail.com>

* fix imagePullSecrets for nfs-provisioner

Signed-off-by: kvaps <kvapss@gmail.com>

* update default rpcbind port for nfs-server-provisioner

Signed-off-by: kvaps <kvapss@gmail.com>

* bump chart version

Signed-off-by: kvaps <kvapss@gmail.com>

kvaps commented:
Not sure If I can sign off for the others

kvaps commented:
/assign @kmova any ideas?

kmova commented:
@kvaps - CLA is tricky. So for the migration PRs from kubernetes-incubator to this new repo, I had to cherry pick the commits and fix the ones without CLA by dropping them and adding a new commit with the changes.

kvaps commented:
/hold

kvaps committed:
Add StorageClass parameters (#7519)

* Add StorageClass parameters

Signed-off-by: Andrew Tarasenko <andrexus@gmail.com>

* update chart version

Signed-off-by: Andrew Tarasenko <andrexus@gmail.com>

* fix lint issues. update chart version

Signed-off-by: Andrew Tarasenko <andrexus@gmail.com>

* update README.md

Signed-off-by: Andrew Tarasenko <andrexus@gmail.com>

kvaps committed:
[stable/nfs-server-provisioner] fix deployment when not creating storage class (#16331)

* When a storage class is not created, mountOptions and parameters are also not included in the file

Signed-off-by: Tomas Ostasevicius <t.ostasevicius@gmail.com>

* Bump nfs-server-provisioner version

Signed-off-by: Tomas Ostasevicius <t.ostasevicius@gmail.com>

kvaps committed:
Upgrade nfs-provisioner to v2.3.0 (#20468)

Signed-off-by: Luis Urraca <lurraca@pivotal.io>

kvaps committed:
stable/nfs-server-provisioner Add ability to specify pod Security Context settings (#22642)

* Add ability to specify Security Context settings for
NFS Server Provisioner.

Signed-off-by: Zoram Thanga <zoram@cloudera.com>

* Changed version number, and changed Helm value to
podSecurityContext to avoid confusion with container level
Security Context.

Signed-off-by: Zoram Thanga <zoram@cloudera.com>

kvaps committed:
statdNodePort is not used (#22660)

Signed-off-by: chenlein <leichen.china@hotmail.com>

kvaps commented:
/unhold

kvaps commented:
> @kvaps - CLA is tricky. So for the migration PRs from kubernetes-incubator to this new repo, I had to cherry pick the commits and fix the ones without CLA by dropping them and adding a new commit with the changes.

Okay seems rebased, any idea if `/deploy/helm` good place for storing the helm chart?

consideRatio commented on deploy/helm/Chart.yaml:
```diff
@@ -0,0 +1,20 @@
+  email: naseem@transit.app
+home: https://github.com/kubernetes/charts/tree/master/stable/nfs-server-provisioner
+sources:
+- https://github.com/kubernetes-incubator/external-storage/tree/master/nfs
```
```suggestion
home: https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner
sources:
- https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner/tree/master/deploy/helm
```

consideRatio commented on deploy/helm/README.md:
```diff
@@ -0,0 +1,206 @@
+# NFS Server Provisioner
+
+[NFS Server Provisioner](https://github.com/kubernetes-incubator/external-storage/tree/master/nfs)
```
```suggestion
[NFS Server Provisioner](https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner)
```

kmova commented on deploy/helm/Chart.yaml:
```diff
@@ -0,0 +1,20 @@
+  email: joaocc-dev@live.com
+- name: naseemkullah
+  email: naseem@transit.app
+home: https://github.com/kubernetes/charts/tree/master/stable/nfs-server-provisioner
```
@kvaps - wondering if we can get these links updated in the current PR itself as an additional commit. Also update the references from kubernetes-incubator to kubernetes-sigs.

kmova commented:
Thanks @kvaps - this looks good. Just had one minor comment. Let me know if you would like to take that up as part of this commit or in another PR.

consideRatio commented:
Thank you for your work on this @kvaps and @kmova!

kvaps committed:
Update deploy/helm/README.md

Co-authored-by: Erik Sundell <erik.i.sundell@gmail.com>

kmova commented:
/lgtm
/approved
